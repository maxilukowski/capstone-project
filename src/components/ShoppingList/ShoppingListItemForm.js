import React, { useRef } from 'react'
import styled from 'styled-components'
import ButtonS from '../Buttons'

const ShoppingListItemForm = ({ onSubmit }) => {
  const input = useRef(null)
  return (
    <StyledForm
      data-testid="test"
      onSubmit={(e) => formSubmit(e, onSubmit, input)}
    >
      <StyledInput
        maxLength="20"
        type="text"
        ref={input}
        name="input"
        placeholder="I need.."
        required
      />
      <ButtonS text="add" fontSize="20px" />
    </StyledForm>
  )
}

export default ShoppingListItemForm

export function formSubmit(event, onSubmit, input) {
  event.preventDefault()
  const title = input.current.value
  onSubmit(title)
  event.target.reset()
}

const StyledInput = styled.input`
  width: 275px;
  margin-right: 10px;
  border-radius: 3px;
  border: none;
  background: var(--sand);
  outline: none;
  font-size: 20px;
`

const StyledForm = styled.form`
  display: flex;
  justify-content: space-between;
`
