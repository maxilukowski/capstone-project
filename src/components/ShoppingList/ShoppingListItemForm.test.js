import React from 'react'
import ShoppingListItemForm, { formSubmit } from './ShoppingListItemForm'
import { fireEvent, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('ShoppingListItemForm.js', () => {
  describe('testing formfields', () => {
    const testString = 'text'
    let input
    let button
    let handleSubmit
    let form
    beforeEach(() => {
      handleSubmit = jest.fn()
      render(<ShoppingListItemForm onSubmit={handleSubmit} />)
      input = screen.getByPlaceholderText('I need..')
      button = screen.getByTestId('shopping-listitem-form-button')
      form = screen.getByTestId('shoppingListItemForm')
    })
    test('inputfield & button should be in the document', () => {
      expect(input).toBeInTheDocument()
      expect(button).toBeInTheDocument()
    })
    test('inputfield should accept text', () => {
      expect(input.value).toBe('')
      fireEvent.change(input, { target: { value: testString } })
      expect(input.value).toBe(testString)
    })
    test('should submit form on buttonClick', () => {
      fireEvent.click(button)
      fireEvent.submit(form)
      expect(handleSubmit).toHaveBeenCalledTimes(2)
    })
  })

  describe('formSubmit function', () => {
    let onSubmit
    let event
    let input
    beforeEach(() => {
      onSubmit = jest.fn()
      event = {
        target: {
          reset: jest.fn(),
        },
        preventDefault: () => {},
      }
      input = {
        current: {
          value: null,
        },
      }
      formSubmit(event, onSubmit, input)
    })
    function toHaveBeenCalledOnce(spyOn) {
      expect(spyOn).toHaveBeenCalledTimes(1)
    }
    test('triggers event.target.reset', () => {
      toHaveBeenCalledOnce(event.target.reset)
    })
    test('triggers onSubmit prop', () => {
      toHaveBeenCalledOnce(onSubmit)
    })
  })
})
