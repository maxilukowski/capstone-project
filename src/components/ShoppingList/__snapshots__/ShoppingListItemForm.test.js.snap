// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingListItemForm.js testing formfields matches snapshot 1`] = `"<form data-testid=\\"test\\" class=\\"sc-AxhCb jdesIU\\"><input maxlength=\\"20\\" type=\\"text\\" name=\\"input\\" placeholder=\\"I need..\\" required=\\"\\" class=\\"sc-AxiKw gELxW\\"><button font-size=\\"20px\\" class=\\"sc-AxjAm ffZWFz\\">add</button></form>"`;
